src/core/or/connection_edge.c

```diff
@@ -863,18 +863,30 @@ export_hs_client_circuit_id(edge_connection_t *edge_conn,
   if (protocol != HS_CIRCUIT_ID_PROTOCOL_HAPROXY)
     return;
 
+  const origin_circuit_t *circ = NULL;
   char *buf = NULL;
   const char dst_ipv6[] = "::1";
   /* See RFC4193 regarding fc00::/7 */
-  const char src_ipv6_prefix[] = "fc00:dead:beef:4dad:";
+  const char src_ipv6_prefix[] = "fc00:";
   uint16_t dst_port = 0;
   uint16_t src_port = 1; /* default value */
   uint32_t gid = 0; /* default value */
+  uint32_t rp_addr = 0;
+  uint16_t rp_port = 0;
 
   /* Generate a GID and source port for this client */
   if (edge_conn->on_circuit != NULL) {
-    gid = TO_ORIGIN_CIRCUIT(edge_conn->on_circuit)->global_identifier;
+    circ = CONST_TO_ORIGIN_CIRCUIT(edge_conn->on_circuit);
+    gid = circ->global_identifier;
     src_port = gid & 0x0000ffff;
```

Comment:
Putting part of the circuit id in the port isn't documented, please document it as part of this patch.

Reply:
Manpage says `All other values in the header can safely be ignored.`

@ahf: Why are we putting part of the circuit id in place of source port? Should I document it, remove it or what?

Reply:
Removing it is a breaking change, if someone is relying on the implementation. (Rather than the documentation.)

So it's safer to document it. If we decide to remove it, we should replace it with part of the fingerprint (so it varies in a similar way), and document that breaking change in the changes file.

Reply:
If we change the source port, we'll also need to update the tests.

Reply:
See the remaining issues list in:
https://trac.torproject.org/projects/tor/ticket/32604#comment:19